### PR TITLE
Added more code validation checkts to CI

### PR DESCRIPTION
**Pull request by bkhadars**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -19,12 +19,37 @@ jobs:
...
         go install golang.org/x/lint/golint
 
-    - name: Run go fmt & lint
+    - name: Run Go fmt
       run: |
-        go fmt -mod=readonly ./...
-        git diff HEAD
-        if [ `git diff HEAD --name-only | wc -l` -gt 0 ]; then exit 1; fi
-        golint -set_exit_status=1 `go list -mod=readonly ./... | grep -v /vendor/`
+        unformatted=$(IS_CONTAINER=FALSE ./hack/go-fmt.sh .)
```
Comment (clnperez) [inline .github/workflows/ci.yml:24]: i'm not sure, but i'm guessing that the issue you saw might be worked around by putting quotes around this expression or something to that effect. did you try any other ways to format this?

**Comment by clnperez** [comment]
any updates? you'd mentioned you were going to look at removing the validate vendor check

**Comment by bkhadars** [comment]
@clnperez update the change, commented vendor check for now

**Comment by clnperez** [comment]
You may need to pull in the latest upstream to get these checks to pass. I just tried re-running them and no-go.

**Comment by clnperez** [comment]
@bkhadars any way you can get this in soon? i'm kind of waiting on it before we check in too many other PRs so we can have these checks.

**Comment by bkhadars** [comment]
Raised PR for verifying codegen script, I think that PR should fix the issue

**Comment by clnperez** [comment]
I think you have to rebase one more time to pick up the codegen fixes.

**Comment by clnperez** [comment]
@bkhadars ping

**Comment by bkhadars** [comment]
@clnperez I have rebased the code to latest, but the checks are still failing  as go vet list errors from the platform validation testcases... We should fix those platform validation test cases..

I have commented out those testcases for time being and raised the New PR with the other vet changes

**Comment by bkhadars** [comment]
@clnperez I have rebased the changes and vet is failing as the vendor changes are required...

**Comment by clnperez** [comment]
the codegen test exits with errors but doesn't fail the check:
```
+ go generate ./pkg/types/installconfig.go
github.com/openshift/installer/pkg/types:-: conflicting types in allOf branches in schema: string vs DiskCategory
github.com/openshift/installer/pkg/types:-: conflicting types in allOf branches in schema: string vs DiskCategory
github.com/openshift/installer/pkg/types:-: conflicting types in allOf branches in schema: string vs DiskCategory
Error: not all generators ran successfully
run `controller-gen crd:crdVersions=v1 paths=. output:dir=../../data/data/ -w` to see all available markers, or `controller-gen crd:crdVersions=v1 paths=. output:dir=../../data/data/ -h` for usage
exit status 1
pkg/types/installconfig.go:70: running "go": exit status 1
+ go generate ./pkg/rhcos/ami.go
```

and it should be in its own job like the rest. 

> and vet is failing as the vendor changes are required...

If we check in all the vendor changes this ups our likelihood for merge conflicts exponentially. We just can't check in those since we pull in upstream changes all the time. If you need the vendored files to be there just do what is done for the build, and run `go mod vendor` first before running the `go vet`. You could even combine those two tests to shorten the time of the checks.

**Comment by clnperez** [comment]
looks like the codegen issue is actually a problem with the test itself (see https://github.com/openshift/installer/pull/5406#issuecomment-984090989) so we can leave that bit as-is

**Comment by clnperez** [comment]
in trying to understand what was going on with your PR I ended up just creating my own b/c that was the only way to test it, so, we can just close this one. thanks for the other fixes and getting this started!

**Pull request closed without merging**